src/Filesystem.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Yiisoft\Files;
+
+use League\Flysystem\Filesystem as LeagueFilesystem;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\PathNormalizer;
+use Yiisoft\Aliases\Aliases;
+
+class Filesystem extends LeagueFilesystem implements FilesystemInterface
+{
+    private Aliases $aliases;
+
+    public function __construct(FilesystemAdapter $adapter, array $aliases = [], array $config = [], PathNormalizer $pathNormalizer = null)
+    {
+        if ($aliases !== []) {
```

Comment:
I think it's better to put it in the method

```php
$this->aliases = new Aliases($this->prepareAliases($aliases));
```

```php
   private function prepareAliases(array $aliases): array {
        if ($aliases !== []) {
            return array_merge(['@root' => ''], $aliases);
        }

        return $aliases;
    }
```

Reply:
Done